newIDE/app/src/SceneEditor/MosaicEditorsDisplay/index.js

```diff
@@ -167,6 +166,20 @@ const MosaicEditorsDisplay = React.forwardRef<
     if (start) editor.restartSceneRendering();
     else editor.pauseSceneRendering();
   }, []);
+  const openNewObjectDialog = React.useCallback(
+    () => {
+      if (!isEditorVisible('objects-list')) {
+        // Objects list is not opened. Open it now.
+        toggleEditorView('objects-list');
+      }
+
+      // Open the new object dialog when the objects list is opened.
+      objectsListDoNowOrAfterRender((objectsList: ?ObjectsListInterface) => {
```

Comment:
are we sure this will always be the first thing to happen on the next render? Isn't a timeout roughly safer?

Reply:
The effect will be run at the next render (and if for some reason the objects list is not there, it won't crash but will just do nothing). It should be there because a state was changed just before, so running the effect at the next render is the safest thing i think. If you use a timeout, you have a slight chance that something else removed the objects list in the meantime.